icu4c/source/i18n/regexcmp.h

```diff
@@ -154,12 +151,6 @@ class U_I18N_API RegexCompile : public UMemory {
     RegexPatternChar              fC;                // Current char for parse state machine
                                                      //   processing.
 
-    //
-    //   Data for the state machine that parses the regular expression.
-    //
-    RegexTableEl                  **fStateTable;     // State Transition Table for regex Rule
```

Comment:
You could also take out the forward declaration of RegexTableEl at line 40 in this file.
The changes to this file otherwise look good - I think that they're all just oversights or sloppy maintenance.